goagen/gen_test/generator.go

```diff
@@ -0,0 +1,299 @@
+	imports := []*codegen.ImportSpec{
+		codegen.SimpleImport("bytes"),
+		codegen.SimpleImport("fmt"),
+		codegen.SimpleImport("net/http"),
+		codegen.SimpleImport("net/http/httptest"),
+		codegen.SimpleImport("testing"),
+		codegen.SimpleImport(AppPkg),
+		codegen.SimpleImport("github.com/goadesign/goa"),
+		codegen.SimpleImport("github.com/goadesign/goa/goatest"),
+		codegen.SimpleImport("golang.org/x/net/context"),
+	}
+
+	return api.IterateResources(func(res *design.ResourceDefinition) error {
+		filename := filepath.Join(codegen.OutputDir, res.Name+".go")
+		file, err := codegen.SourceFileFor(filename)
+		if err != nil {
+			return err
+		}
+		if err := file.WriteHeader("", "test", imports); err != nil {
+			return err
+		}
+
+		var methods = []TestMethod{}
+
+		for _, action := range res.Actions {
+			for _, response := range action.Responses {
+				if response.Name == design.SwitchingProtocols {
+					continue
+				}
+				for routeIndex, route := range action.Routes {
+					mediaType := design.Design.MediaTypeWithIdentifier(response.MediaType)
+					if mediaType == nil {
+						methods = append(methods, createTestMethod(res, action, response, route, routeIndex, nil, nil))
+					} else {
+						for _, view := range mediaType.Views {
+							methods = append(methods, createTestMethod(res, action, response, route, routeIndex, mediaType, view))
+						}
+					}
+				}
+			}
+		}
+
+		g.genfiles = append(g.genfiles, filename)
+		err = resourceTestContextTmpl.Execute(file, methods)
+		if err != nil {
+			panic(err)
+		}
+		return file.FormatCode()
+	})
+}
+
+func createTestMethod(resource *design.ResourceDefinition, action *design.ActionDefinition, response *design.ResponseDefinition, route *design.RouteDefinition, routeIndex int, mediaType *design.MediaTypeDefinition, view *design.ViewDefinition) TestMethod {
+	routeNameQualifier := suffixRoute(action.Routes, routeIndex)
+	viewNameQualifier := func() string {
+		if view != nil && view.Name != "default" {
+			return view.Name
+		}
+		return ""
+	}()
+	method := TestMethod{}
+	method.Name = fmt.Sprintf("%s%s%s%s%s", codegen.Goify(action.Name, true), codegen.Goify(resource.Name, true), codegen.Goify(response.Name, true), routeNameQualifier, codegen.Goify(viewNameQualifier, true))
+	method.ActionName = codegen.Goify(action.Name, true)
+	method.ResourceName = codegen.Goify(resource.Name, true)
+	method.Comment = fmt.Sprintf("test setup")
+	method.ControllerName = fmt.Sprintf("%s.%sController", TargetPackage, codegen.Goify(resource.Name, true))
+	method.ContextVarName = fmt.Sprintf("%sCtx", codegen.Goify(action.Name, false))
+	method.ContextType = fmt.Sprintf("%s.New%s%sContext", TargetPackage, codegen.Goify(action.Name, true), codegen.Goify(resource.Name, true))
+	method.RouteVerb = route.Verb
+	method.Status = response.Status
+	method.FullPath = goPathFormat(route.FullPath())
+
+	if view != nil && mediaType != nil {
```

Comment:
Not based on this; https://github.com/goadesign/goa/pull/422/files/0b8212997702aac0b1a3e0e584899bcc6ea6c53b#diff-74f47f25c782559a460b41b7cc7e97a8R112

And as far as I can tell I need both to find the correct type information.. ?